src/main/java/startsteps/ECommerceShop/service/CartService.java

```diff
@@ -98,4 +96,29 @@ private List<CartProductResponse> createCartProduct(List<CartProduct> cartProduc
                 cartProduct.getQuantity(),
                 cartProduct.getPrice())).collect(Collectors.toList());
         }
+
+    @Transactional
+    public CartResponse getCart(User user){
+        Cart cart = user.getCart();
+
+        if (cart == null || cart.getCartProductList().isEmpty()){
+            throw new EmptyCartException("You haven't added anything yet, your cart is empty");
+        }
+        log.info("get cart: {}", user.getCart());
```

Comment:
you can pass in `cart` that you defined on line 102

Reply:
done